File: LeetCode/Simulation/Golang/Leetcode2351-first-letter-to-appear-twice.go
package sbgo

func repeatedCharacter(s string) byte {
	ans, st := byte(0), make([]int, 26)
	for _, x := range s {
		if st[x-'a'] == 1 {
			ans = byte(x)
			break
		}
		st[x-'a']++
	}
	return ans
}

File: LeetCode/Simulation/Rust/Leetcode2351-first-letter-to-appear-twice.rs
impl Solution {
  pub fn repeated_character(mut s: String) -> char {
    let (mut ans, mut st) = ('z', vec![0; 26]);
    for x in s.as_bytes() {
      if st[(x - b'a') as usize] == 1 {
        ans = x.clone() as char;
        break;
      }
      st[(x - b'a') as usize] += 1;
    }
    ans
  }
}
